src/fu-security-attr.c

```diff
@@ -247,3 +250,61 @@ fu_security_attr_get_result(FwupdSecurityAttr *attr)
 	/* TRANSLATORS: Suffix: the fallback HSI result */
 	return _("Failed");
 }
+
+gchar *
+fu_security_attrs_to_json_string(FuSecurityAttrs *attrs)
+{
+	/*  Expected Seccurity attr format
+	{
+	    "SecurityAttrs": {
+		"Attrs": [
+		    {
+			"name": "aaa",
+			"value": "bbb"
+		    }
+		]
+	    }
+	}
+	*/
+	gchar *data = NULL;
+	JsonNode *json_root = NULL;
+	JsonGenerator *json_generator = NULL;
+	JsonBuilder *builder = json_builder_new();
+	fu_security_attrs_to_json(attrs, builder);
+	/* export as a string */
+	json_root = json_builder_get_root(builder);
+	json_generator = json_generator_new();
+	json_generator_set_pretty(json_generator, TRUE);
+	json_generator_set_root(json_generator, json_root);
+	data = json_generator_to_data(json_generator, NULL);
+	if (data == NULL) {
+		g_warning("Converting to Json string error");
```

Comment:
rather than a warning, you should pass error into this function and set an error if this scenario happens (`data` is NULL) and let the caller handle it.

Reply:
Thanks. I'll add this part of the codes.